File: src/Hypnonema.Shared/Models/Schedule.cs
namespace Hypnonema.Shared.Models
{
    using System;
    
    public class Schedule
    {
        public DayOfWeek? DayOfWeek { get; set; }

        public DateTime EndDate { get; set; }

        public DateTime StartDateTime { get; set; }

        public int Id { get; set; }

        public int Interval { get; set; }

        public int Rule { get; set; }

        public Screen Screen { get; set; }

        public string Url { get; set; }
    }
}

File: src/Hypnonema.NUI/src/types/schedule.ts
import Screen from "./screen";

export enum Recurrence {
  Monthly = 1,
  Weekly,
  Daily,
  Hourly,
}

export enum DayOfWeek {
  Monday,
  Tuesday,
  Wednesday,
  Thursday,
  Friday,
  Saturday,
  Sunday,
}

export default interface Schedule {
  id?: number;
  startDateTime: string;
  endDate?: string;
  rule: Recurrence;
  interval: number;
  dayOfWeek?: DayOfWeek;
  url: string;
  screen: Screen;
}
